velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -207,6 +229,42 @@ VeloxToSubstraitExprConvertor::toSubstraitLiteral(
   return *literalExpr;
 }
 
+void VeloxToSubstraitExprConvertor::complexVectorToLiteral(
+    google::protobuf::Arena& arena,
+    std::shared_ptr<ConstantVector<ComplexType>> constantVector,
+    ::substrait::Expression_Literal* substraitLiteral) {
+  VELOX_CHECK_EQ(
+      constantVector->size(), 1, "Only one complex vector is expected.");
+  auto arrayVector = constantVector->valueVector()->as<ArrayVector>();
+  VELOX_CHECK_NOT_NULL(
+      arrayVector,
+      "arrayVector is NULL, Maybe Substrait literal is not supported for {}.",
+      constantVector->type()->toString());
+  VELOX_CHECK_EQ(arrayVector->size(), 1, "Only one array is expected.");
+  if (constantVector->isNullAt(0)) {
+    // Process the null value.
+    substraitLiteral->MergeFrom(exprConvertor_->toSubstraitNullLiteral(
+        arena, arrayVector->type()->kind()));
+  } else {
+    ::substrait::Expression_Literal_List* listLiteral =
+        google::protobuf::Arena::CreateMessage<
+            ::substrait::Expression_Literal_List>(&arena);
+    if (arrayVector->elements()->isScalar()) {
```

Comment:
This code needs to use ConstantVector::index() to convert a specific row of the array vector.

Reply:
Do you mean use the code like  "constantVector->valueAt(constantVector->index())" to get a specific row of the array vector.
In fact, I found variable value_ in class constantVector is not assigned. So we should still use arrayVector->elements() here.  Please guide me if I misunderstand your comments.
![image](https://user-images.githubusercontent.com/14065538/197964687-9bea15ff-966a-422e-b843-3db0565418f2.png)

Reply:
To create a constant vector a non-zero index,

```
   auto arrayVector = makeArrayVector({
      {1, 2, 3},
      {4, 5},
   }),

// c represett {4, 5] array, not [1. 2, 3] and not {1, 2, 3, 4, 5}
auto c = BaseVector::wrapInConstant(10, 1, arrayVector);
```

Reply:
Thank you, I will try it.